### PR TITLE
test checkout.com api

### PR DESCRIPTION
**Pull request by aryan159**
To test, navigate to the checkout page and press "Request Payment".
You will have to install "yarn add checkout-sdk-node"

### CODE REVIEW

**Comment by aryan159** [comment]
The full integration of the "Request a Payment" api route has been tested. Here's how it goes

1. In the checkout page, enter your card number (sample: 4242 4242 4242 4242), cvv (sample: 100) and expiry date (any future date).
2. This will get tokenised and sent to our server
3. The server will call the checkout.com api
4. The server will return the response to the client to handle

You can see that the payment gets logged in the [test dashboard](https://dashboard.sandbox.checkout.com) under payments.
Login Username: aryan.jain@u.nus.edu
Login Password : JuniorsClub1!

TODO
1. fix deploy error occurring due to the use of synchronous script loading
2. fix page formatting
3. Clean up code and make it more reusable

**Comment by kevinchua6** [comment]
Good job! It's working for me. A thing to note is that your iframes don't appear when I'm switching between the links at the top, probably because of your `<script>` tags

File: components/Layout.tsx [hunk 1]
```diff
@@ -27,7 +27,14 @@ const Layout = ({ children, title = 'This is the default title' }: Props) => (
         <Link href='/users'>
           <a>Users List</a>
         </Link>{' '}
-        | <Link href='/api/users'>Users API</Link>
+        |{' '}
+        <Link href='/api/users'>
+          <a>Users API</a>
+        </Link> {' '}
+        |{' '}
+        <Link href='/checkout'>
```
Comment (MorningLit) [inline components/Layout.tsx:35]: use a normal a link for this because we want a full page refresh instead of client side routing

File: tsconfig.json [hunk 2]
```diff
@@ -23,7 +23,7 @@
     "next-env.d.ts",
     "**/*.ts",
     "**/*.tsx"
-  ],
+, "pages/api/checkout/index.tsx"  ],
```
Comment (MorningLit) [inline tsconfig.json:26]: why is this here?

File: pages/api/checkout/index.tsx [hunk 3]
```diff
@@ -0,0 +1,47 @@
+/* eslint-disable import/no-anonymous-default-export */
+import { Checkout } from 'checkout-sdk-node';
+import { NextApiRequest, NextApiResponse } from "next";
+export default async (req: NextApiRequest, res: NextApiResponse) => {
```
Comment (MorningLit) [inline pages/api/checkout/index.tsx:4]: use export default async function handler to remove the eslint-disable at the top of the page

File: pages/api/checkout/index.tsx [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+	console.log(req);
+  const token = req.query.token;
+
+  const cko = new Checkout('sk_sbox_4qlnxtd6xro7e4hjhpuyyq7ixa4', { pk: 'pk_sbox_5cdu3zk4ywwlclah2kcfiq7p2qt'});
```
Comment (MorningLit) [inline pages/api/checkout/index.tsx:8]: put secret keys and public keys in .env

**Review by MorningLit — COMMENTED** [review]
Kevin asked me to review your PR even though it was still in draft mode, I left some small comments for you, no rush, btw nicely done PR! 👍

**Review by kevinchua6 — APPROVED** [review]
